File: Problems/ICPC Asyut Sheets/Sheet #2 (Loops)/J. Primes from 1 to n.cpp
#include <bits/stdc++.h>
using namespace std;

bool isPrime(int x)
{
    if (x < 2 || (x % 2 == 0 && x != 2))
        return false;

    for (int i = 2; i <= x / 2; i++)
        if (x % i == 0)
            return false;

    return true;
}
int main()
{
    int n;
    cin >> n;
    for (int i = 2; i <= n; i++)
        if (isPrime(i))
            cout << i << " ";

    return 0;
}

File: Problems/ICPC Asyut Sheets/Sheet #2 (Loops)/J. Primes from 1 to n.py
def isPrime(n:int)->bool:
    if n<2:return False
    if n==2: return True
    if n%2==0: return False
    for i in range(2,n//2+1):
        if n%i==0: return False
    return True

n = int(input())
for i in range(1,n+1):
    if isPrime(i): print(i,end=' ')
print()